packages/graphql/src/schema/resolvers/subscriptions/utils/compare-properties.ts

```diff
@@ -154,19 +151,51 @@ function parseFilterProperty(key: string): { fieldName: string; operator: string
     return { fieldName, operator };
 }
 
+const multipleConditionsAggregationMap = {
+    AND: (results: boolean[]): boolean => {
+        for (const res of results) {
+            if (!res) {
+                return false;
+            }
+        }
+        return true;
+    },
+    OR: (results: boolean[]): boolean => {
+        for (const res of results) {
+            if (res) {
+                return true;
+            }
+        }
+        return false;
+    },
+};
+
 /** Returns true if receivedProperties comply with filters specified in whereProperties, false otherwise. */
 export function filterByProperties<T>(
     node: Node,
-    whereProperties: Record<string, T>,
+    whereProperties: Record<string, T | Array<Record<string, T>>>,
     receivedProperties: Record<string, T>
 ): boolean {
     for (const [k, v] of Object.entries(whereProperties)) {
-        const { fieldName, operator } = parseFilterProperty(k);
-        const receivedValue = receivedProperties[fieldName];
-        const fieldMeta = node.primitiveFields.find((f) => f.fieldName === fieldName);
-        const checkFilterPasses = getFilteringFn(operator);
-        if (!checkFilterPasses(receivedValue, v, fieldMeta)) {
-            return false;
+        if (Object.keys(multipleConditionsAggregationMap).includes(k)) {
+            const comparisonResultsAggregationFn = multipleConditionsAggregationMap[k];
+            if (!comparisonResultsAggregationFn) {
+                return false;
+            }
```

Comment:
Could this happen? or is it just typescript complaining?

Reply:
cannot happen, I used to check with hardcoded "AND", "OR" on line 180 but then changed to Object.keys. Will remove this check :D